### PR TITLE
Fixed error with uid

### PR DESCRIPTION
**Pull request by roigcarlo**
This PR fixes https://github.com/simphony/simphony-common/issues/98

### CODE REVIEW

**Comment by itziakos** [comment]
We probably need some tests to make sure that we will not have a regression of the issue

**Comment by roigcarlo** [comment]
@itziakos I have extended the iterator tests to check that as is where was present. I'm not sure if this is the proper way to check it, but I am not able to find any other way to verify that the two sets of id's are the same.

 I have also added a missing test to check point_iter() without arguments.

File: simphony/io/tests/test_h5_mesh.py [hunk 1]
```diff
@@ -332,6 +350,15 @@ def test_get_all_edges_iterator(self):
 
         self.assertItemsEqual(iedges_id, euids)
 
+        mesh_points = [p.uid for p in self.mesh.iter_points()]
```
Comment (itziakos) [inline simphony/io/tests/test_h5_mesh.py]: We should check that the specific uuids in each edge.points are equal to the uuids there where given at initialization. 
something like
```python
for index, edge in self.mesh.iter_edges(euids):
    self.assertItemEqual(edge.points, edges[index].points)
```

Comment (itziakos) [inline simphony/io/tests/test_h5_mesh.py]: Similar logic can be applied in the other places.

**Comment by itziakos** [comment]
> @itziakos I have extended the iterator tests to check that as is where was present. I'm not sure if this is the proper way to check it, but I am not able to find any other way to verify that the two sets of id's are the same.

The check needs to be more specific.

**Comment by roigcarlo** [comment]
>  something like
>  ```python
>  for index, edge in self.mesh.iter_edges(euids):
>      self.assertItemEqual(edge.points, edges[index].points)
>  ```

Actually, I was trying to avoid this type of check,

As far as I know python iterators were not designed to be indexed, so what comes into my mind to implemented this kind of check is something like:
```python
for index, edge in enumerate(self.mesh.iter_edges(euids)):
    self.assertItemsEqual(edge.points, edges[index].points)
```

Which works fine for this particular case. However, enumerating the iterators "on the fly" makes me wonder if the order in which the elements are going to be iterated and enumerated is going t be the same as the insertion order. Is it right?

**Comment by itziakos** [comment]
> As far as I know python iterators were not designed to be indexed, so what comes into my mind to implement this kind of check is something like:

yes, sorry that was a stupid error, using enumerate was the intention

> Which works fine for this particular case. However, enumerating the iterators "on the fly" makes me wonder if the order in which the elements are going to be iterated and enumerated is going t be the same as the insertion order. Is it right?

Yes, so far, I have been working and implementing code under this assumption that when we provide a value of the uids argument we get the data in order. But you make a good point that we should mention that in the docstrings.

**Comment by itziakos** [comment]
:+1:

**Pull request merged**